tests/lit-tests/signed-integer-overflow.ispc

```diff
@@ -0,0 +1,54 @@
+// This test checks for the use of the no signed wrap (`nsw`) attribute for arithmetic
+// expressions which may suffer from overflow.
+// Needed for induction variable simplification dicussed in
+// [GitHub issue ISPC/#2460](https://github.com/ispc/ispc/issues/2460).
+
+// RUN: %{ispc} %s -o %t.ll -O2 --pic --emit-llvm-text --target=avx2 --arch=x86-64
+// RUN: FileCheck %s --input-file=%t.ll --check-prefix=CHECK_NSW
+
+// CHECK_NSW-LABEL:         define {{.*}} @mul_u
+// CHECK_NSW-COUNT-2:         {{.*}}= mul {{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @mul_v
+// CHECK_NSW-COUNT-2:         {{.*}}= mul {{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @add_u
+// CHECK_NSW-COUNT-2:         {{.*}}= add {{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @add_v
+// CHECK_NSW-COUNT-2:         {{.*}}= add {{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @sub_u
+// CHECK_NSW-COUNT-2:         {{.*}}= sub {{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @sub_v
+// CHECK_NSW-COUNT-2:         {{.*}}= sub {{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @shl_u
+// CHECK_NSW-COUNT-1:         {{.*}}= shl {{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @shl_v
+// CHECK_NSW-COUNT-1:         {{.*}}= shl {{.*}}nsw
+// CHECK_NSW-LABEL:         define {{.*}} @neg_u
+// CHECK_NSW-NOT:             nsw
+// CHECK_NSW-LABEL:         define {{.*}} @neg_v
+// CHECK_NSW-NOT:             nsw
+// CHECK_NSW-LABEL:         define {{.*}} @ptrMath
+// CHECK_NSW-NOT:             nsw
+
+// REQUIRES: X86_ENABLED
+
+unmasked void mul_u(uniform int a, uniform int b, uniform int *uniform c, uniform int k) { c[0] = a * k * b; }
```

Comment:
Why writing to a pointer instead of just returning the value?

Reply:
To mimic how values would be returned if using the ISPCRT API where are values generally returned via pointers. If you prefer return here, I can change it.

Reply:
I would prefer to have small and clean example, which is not complicated with other effects. For more complex stuff, you can check it separately.